examples/complex/xss_scanner.py

```diff
@@ -380,20 +380,21 @@ def handle_starttag(self, tag, attrs):
                            injection_point,
                            " onmouseover=alert(0) t=",
                            match.decode('utf-8'))
-        elif in_HTML and not in_script and inject_open_angle and inject_close_angle and inject_slash:  # e.g. <html>PAYLOAD</html>
+        elif in_HTML_val and not in_script_val and inject_open_angle and inject_close_angle and inject_slash:  # e.g. <html>PAYLOAD</html>
             return XSSData(request_URL,
                            injection_point,
                            '<script>alert(0)</script>',
                            match.decode('utf-8'))
         else:
             return None
+    return None
 
 
 # response is mitmproxy's entry point
 def response(flow: http.HTTPFlow) -> None:
     cookiesDict = get_cookies(flow)
     # Example: http://xss.guru/unclaimedScriptTag.html
-    find_unclaimed_URLs(flow.response.content, flow.request.url)
+    find_unclaimed_URLs(flow.response.content.decode('utf-8'), flow.request.url)
```

Comment:
This will fail for any non-utf8 contents, take a look at `response.get_text` :)

Reply:
Hi, we need to rewrite most of the code since there are many places where we are taking that the content will be utf8 only. `response` function has been written on this assumption so is most of the code. I thought this assumption was wrong but since this was an already existing code didn't ask.
@mhils I will go forward and check all of the code, it would be great if you could review the whole existing code once. Thank you